### ISSUE TITLE
Raw diff between tags and branches

### ISSUE THREAD

**Issue by pfactum** [go-gitea/gitea#13416]
Currently, it is possible to get a raw diff between 2 commits, for instance:

https://try.gitea.io/Etzelia/review/commit/4be8c56ae08e83c486854d9221cbd16e84607ca8...2eaa86164eb4cca84e14be1c357398c7b2c4efa2.diff

The same doesn't work, however, if tags are used instead of commit IDs, and branch names does not seem to work either. For instance:

https://try.gitea.io/ikervagyok/reflex-platform/commit/v0.5.3.0...v0.6.0.0.diff

It would be nice to have some unified URL to get a plain diff between any tags, commits and branch HEADs mix.

An example of how this works with github:

https://github.com/pfactum/pf-kernel/compare/v5.9...pf-5.9.diff

This gives a raw diff between a tag and a branch HEAD.

**Comment by pfactum** [comment]
Copy-pasting reasoning from Codeberg Matrix room.

---

to put it in a context, this is a nice feature to use with huge projects like linux kernel where there's some base (say, v5.18 tag), and a small amount of commits on top of that tag. you wanna build the whole kernel, and you already have that base downloaded as a tarball from the kernel.org site, and what you need to download is that small diff on top of the tag without having to clone the whole repo

and this becomes especially useful if cloud-based build system (like opensuse build service) is used. once you rebuild the kernel, you upload just a tiny diff between two small diffs, not a 150 MiB tarball each time

**Comment by pfactum** [comment]
@FWDekker I don't think this is solved. Yes, one can compare between two refs, but there's no way to generate a plain diff between those, which is the point of this RFE. Appending `.diff` to the URL you've mentioned generates error 404.

**Comment by FWDekker** [comment]
Oops, you're right! I missed the "raw" part in your issue description.

**Comment by bircni** [comment]
This has been implemented in #37632 (merged, targeting 1.27.0), which adds raw diff/patch endpoints for repository comparisons between arbitrary refs — including tags and branch HEADs — e.g. /{owner}/{repo}/compare/v0.5...main.diff. Closing as resolved; thanks for the request!
